File: Unit - 1/Brute Force Strategy/Java Implementation/SelectionSort.java
//Program to sort a array using selection sort
import java.util.Arrays;

public class SelectionSort {
    private static void sort(int[] arr){
        for(int i = 0; i < arr.length - 1; i++){
            int min = i;
            for(int j = i + 1;  j < arr.length; j++){
                if(arr[min] > arr[j]){
                    min = j;
                }
            }
            if(i != min){
                swap(arr, i, min);
            }
        }
    }  

    private static void swap(int[] arr, int index1, int index2){
        int temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
    }

    public static void main(String[] args) {
        int[] arr = {56, 32, 90, 65, 10, 34};
        System.out.println("Array before sorting:\n" + Arrays.toString(arr));
        sort(arr);
        System.out.println("\nArray after sorting:\n" + Arrays.toString(arr));
    }
}

File: Unit - 1/Brute Force Strategy/C implementation/selectionSort.c
//Program to sort a array using selection sort
#include<stdio.h>
#define SIZE 50

void readArray(int arr[], int n)
{
    int i;
    for(i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }
}

void printArray(int arr[], int n)
{
    int i;
    for(i = 0; i < n; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

void swap(int *num1, int *num2)
{
    int temp = *num1;
    *num1 = *num2;
    *num2 = temp;
}

void selectionSort(int arr[], int n)
{
    int i, j, min;
    for(i = 0; i < n - 1; i++)
    {
        min = i;        
        for(j = i + 1; j < n; j++)
        {
            if(arr[min] > arr[j]){
                min = j;
            }
        }
        if(min != i)
            swap(&arr[min], &arr[i]);
    }
}

int main()
{
    int n, arr[SIZE];
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    printf("Enter %d elements: ", n);
    readArray(arr, n);
    printf("Array before sorting:\n");
    printArray(arr, n);
    selectionSort(arr, n);
    printf("\nArray after sorting:\n");
    printArray(arr, n);
    return 0;
}